cmd/acra-translator/common/network.go

```diff
@@ -18,55 +18,47 @@ package common
 
 import (
 	"context"
-	"net"
-
-	"github.com/cossacklabs/acra/logging"
-	"github.com/cossacklabs/acra/network"
+	logging "github.com/cossacklabs/acra/logging"
 	"github.com/prometheus/client_golang/prometheus"
+	"net"
 )
 
 // AcceptConnections return channel which will produce new connections from listener in background goroutine
-func AcceptConnections(parentContext context.Context, connectionString string, errCh chan<- error) (<-chan net.Conn, error) {
+func AcceptConnections(parentContext context.Context, listener net.Listener, errCh chan<- error) (<-chan net.Conn, error) {
 	logger := logging.GetLoggerFromContext(parentContext)
-	listenContext, cancel := context.WithCancel(parentContext)
 	connectionChannel := make(chan net.Conn)
-	listener, err := network.Listen(connectionString)
-	if err != nil {
-		cancel()
-		return nil, err
-	}
 
-	// run goroutine that just accept connections and return them and stop on error. you can stop it by closing listener
+	// Run goroutine that just accept connections and return them and stop on error.
+	// We need to have a control under this goroutine while shutdown process but we can't rely on
+	// parentContext.Done since this event comes later that closing listener that accepts (we use listener
+	// provided by caller, and caller is responsible for closing it).
+	// For this reason we assume that *net.OpError == "accept" in case of accepting error
+	// is normal case when shutdown is invoked
```

Comment:
So... If understand correctly, previously closing the listener resulted in an `accept` error being sent into `errCh`, and now we avoid that. However, the context is basically useless now. Well... This is a weird flex. I don't see any immediate way to improve it, but I don't like it being counterintuitive: usually methods which accept a context can be expected to be cancelable via that context. This is not the case here.

Reply:
This is an afford to handle a case of `"use of closed network connection"` error. Yes, context is almost useless, except that we extract logger from it (which is also not obvious for me - why)

Reply:
@ilammy, now I can't reproduce the `use of closed network connection` error with the EE translator. Probably it was due to some intermediate version of code while debugging, so now I think we can avoid it and use just:
```
conn, err := listener.Accept()
if err != nil {
	logger.WithField(logging.FieldKeyEventCode, logging.EventCodeErrorCantAcceptNewConnections).WithError(err).Errorln("Error on accept connection")
	errCh <- err
	return
}
```
which is much more intuitive